### PR TITLE
Feanil/nat monitor

### PR DESCRIPTION
**Pull request by feanil**

### CODE REVIEW

**Comment by feanil** [comment]
@jarv @e0d, Please review.

I'll be squashing this before merging.

File: playbooks/roles/jenkins_admin/files/nat-monitor.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+#        "ec2:StopInstances"
+#      ],
+#      "Effect": "Allow",
+#      "Resource": "*"
+#    }
+#  ]
+# }
+
+# Get the primary NAT instance's IP
+PRIMARY_NAT_IP=`aws ec2 describe-instances --instance-ids $PRIMARY_NAT_ID | jq -r ".Reservations[].Instances[].PrivateIpAddress"`
```
Comment (e0d) [inline playbooks/roles/jenkins_admin/files/nat-monitor.sh:57]: Can't the three api calls for describe-instances be combined?

File: playbooks/roles/jenkins_admin/files/nat-monitor.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+echo `date` "-- Running NAT monitor"
+while [ . ]; do
+  # Check the health of both instances.
+  primary_pingresult=`ping -c $Num_Pings -W $Ping_Timeout $PRIMARY_NAT_IP| grep time= | wc -l`
+  
+  if [ "$primary_pingresult" == "0" ]; then
+    backup_pingresult=`ping -c $Num_Pings -W $Ping_Timeout $BACKUP_NAT_IP| grep time= | wc -l`
+    if [ "$backup_pingresult" == "0" ]; then
+      echo `date` "-- Both NAT devices un reachable."
+      exit 1
```
Comment (e0d) [inline playbooks/roles/jenkins_admin/files/nat-monitor.sh]: An option would be to do cli sns calls.

File: playbooks/roles/jenkins_admin/tasks/nat_monitor.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+- fail: msg="NAT_MONITORS is not defined."
+  when: NAT_MONITORS is not defined
+
+- name: upload the monitor script
+  copy:
+    dest="{{ jenkins_admin_scripts_dir }}/nat-monitor.sh"
+    src="nat-monitor.sh"
+    owner="{{ jenkins_user }}"
+    group="{{ jenkins_group }}"
+    mode="755"
```
Comment (e0d) [inline playbooks/roles/jenkins_admin/tasks/nat_monitor.yml:26]: world perms?

**Comment by e0d** [comment]
Generally looks good to me, can you provide sample config to make it go?

File: playbooks/roles/jenkins_admin/files/nat-monitor.sh (hunk 1 above)
Comment (feanil) [inline playbooks/roles/jenkins_admin/files/nat-monitor.sh:57]: I'm not sure I fully understand what you mean.  Above the describe instances is getting the ID of the backup where as the calls here are getting the Private IPs of the instances.  You mean look up each instance once and then grab the IP and ID from same JSON?

Comment (e0d) [inline playbooks/roles/jenkins_admin/files/nat-monitor.sh:57]: Yes, I meant could you grab them both from the same JSON.

File: playbooks/roles/jenkins_admin/files/nat-monitor.sh (hunk 2 above)
Comment (jarv) [inline playbooks/roles/jenkins_admin/files/nat-monitor.sh]: If you want to simply copy what I did for the s3 sync script see the `onerror()` function that is used to send mail via sns on any error https://github.com/edx/configuration/blob/master/playbooks/roles/aws/templates/send-logs-to-s3.j2#L63  you can also add a `trap onerror ERR SIGHUP SIGINT SIGTERM` on any signal or any time a child process of your script has a non-zero exit code.

File: playbooks/roles/jenkins_admin/files/nat-monitor.sh (hunk 1 above)
Comment (feanil) [inline playbooks/roles/jenkins_admin/files/nat-monitor.sh:57]: I tried to do this but I wasn't able to find a good way to pull out the ip by instance_id using jq, which is what I would need to do.

**Comment by jarv** [comment]
:+1:

**Pull request merged**